tiledb/sm/serialization/array_schema.cc

```diff
@@ -299,7 +299,7 @@ Status dimension_from_capnp(
     Buffer domain_buffer;
     RETURN_NOT_OK(
         utils::copy_capnp_list(domain_reader, dim_type, &domain_buffer));
-    RETURN_NOT_OK((*dimension)->set_domain(domain_buffer.data()));
+    RETURN_NOT_OK((*dimension)->set_domain(domain_buffer.data(), false));
```

Comment:
For other places where this is needed, we make an _unsafe version. Let's match that pattern for consistency?

Reply:
👍 done!